backend/services/userProfile.service.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable func-names */
+const mongoose = require('mongoose');
+const UserProfile = require('../models/userProfile.model');
+const modificationLogService = require('./modificationLog.service');
+const DatabaseError = require('../errors/database.error');
+const DomainError = require('../errors/domain.error');
+
+const UserProfileService = {};
+
+UserProfileService.createUser = async function (userProfileData, authorEmail) {
+    
+    let newRecord = null;
+    
+    if(Object.prototype.hasOwnProperty.call(userProfileData, "signupEmail") &&
+        userProfileData.signupEmail) {
+        userProfileData.signupEmail = userProfileData.signupEmail.toLowerCase();
+    }
+
+    await mongoose.connection.transaction(async () => {
+    
+        try 
```

Comment:
please reformate the code from
```
try
{
// code to execute
}
catch(error)
{
// throw error
}
```
to 
```
try {
    // code to execute
} catch (error) {
    // throw error
}
```
Take a look at examples on [MDN try...catch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch).
Btw, do you know we have the prettier installed in the project? Probably you just need to set it up.